### ISSUE TITLE
Support import of GlyphsApp files

### ISSUE THREAD

**Issue by n8willis** [trufont/trufont#145]
Using glyphs2ufo [1] it should be possible to import files Glyphs projects.

1 - https://github.com/googlei18n/glyphs2ufo

**Comment by adrientetar** [comment]
Yes. Though it seems to be ufo2/python2 atm and introduces a hard dependency to robofab which we are willing to replace with ufoLib.

I don't think I'll work on that myself in the near future.

**Comment by davelab6** [comment]
I see this as an ideal extension to develop sooner rather than later. While
it would be convenient, until the extensions system is ready, its possible
to run glyphs2ufo outside the application and then open the generated UFOs.
A note about the possibility of doing this would be great to include in the
FAQ or official documentation.

**Comment by moyogo** [comment]
Loading a designspace would be a prerequisite for that.
@adrientetar How can we help with that?

**Comment by adrientetar** [comment]
Support of .glyphs files: I don't think I want to make robofab a mandatory dependency again. I think extractor should be refactored onto something much more modular where you can register a file extension just like you can register a representation factory in defcon. Then you can do something like a glyphs plugin.

**Comment by moyogo** [comment]
It makes sense to put support that in extractor. I guess I’m more interested in supporting designspace in trufont ;-)

**Comment by adrientetar** [comment]
Okay. That's pretty tangent to this issue. It would be using the Unified Font Set format proposal.

I don't have any specific ideas for designspace, feel free to make proposals.

**Comment by moyogo** [comment]
OK, I’ll make a proposal. It’s a prerequiste since glyphLib basically generates UFOs and a designspace when the source Glyphs file has masters and instances defined.

**Comment by adrientetar** [comment]
sgtm

**Comment by brawer** [comment]
Sounds like a very big change... In the meantime, could TruFont perhaps import *.glyphs files that contain just a single master? (If it contains multiple masters, perhaps show a dialog so the user can choose which one to import?)

**Comment by adrientetar** [comment]
I see the UFOs generated by glyphslib use the proprietary robofont attributes/storage to store layers, guidelines, lib keys.

**Comment by brawer** [comment]
Oh, in that case, can you [report bugs to glyphsLib](https://github.com/googlei18n/glyphsLib/issues)? Normally I’d just do this myself, but you’ll know better than me what exactly should be changed in glyphsLib.

**Comment by felipesanches** [comment]
I would love to see TruFont support reading and writing glyphs files. ;-)
